### PR TITLE
feat: (pools) Add token amount validations to pool stake/unstake modal

### PR DESCRIPTION
**Pull request by memoyil**
To review: 

https://deploy-preview-2045--pancakeswap-dev.netlify.app/

To reproduce:

1. Go to pools
2. Go to unstake or stake one of the pools that you have
3. Put an amount higher than you have
4. See there is no warning and confirm button is enabled

### CODE REVIEW

File: src/views/Pools/components/PoolCard/Modals/StakeModal.tsx [hunk 1]
```diff
@@ -91,9 +92,15 @@ const StakeModal: React.FC<StakeModalProps> = ({
...
-      setHasReachedStakedLimit(fullDecimalStakeAmount.plus(userData.stakedBalance).gt(stakingLimit))
+    const fullDecimalStakeAmount = getDecimalAmount(new BigNumber(stakeAmount), stakingToken.decimals)
+
+    if (isRemovingStake) {
+      setUserNotEnoughToken(userData.stakedBalance.lt(fullDecimalStakeAmount))
+    } else {
+      if (stakingLimit.gt(0)) {
+        setHasReachedStakedLimit(fullDecimalStakeAmount.plus(userData.stakedBalance).gt(stakingLimit))
+      }
+      setUserNotEnoughToken(userData.stakingTokenBalance.lt(fullDecimalStakeAmount))
```
Comment (hachiojidev) [inline src/views/Pools/components/PoolCard/Modals/StakeModal.tsx:103]: I don't think we need a new state object for this. We can simply calculate whether the value entered is greater than the balance in each render.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Pools/components/PoolCard/Modals/StakeModal.tsx (hunk 1 above)
Comment (memoyil) [reply src/views/Pools/components/PoolCard/Modals/StakeModal.tsx]: Done

**Review by Chef-Cheems — APPROVED** [review]

File: src/views/Pools/components/PoolCard/Modals/StakeModal.tsx [hunk 2]
```diff
@@ -76,6 +76,13 @@ const StakeModal: React.FC<StakeModalProps> = ({
     }
     return stakingLimit.gt(0) && stakingTokenBalance.gt(stakingLimit) ? stakingLimit : stakingTokenBalance
   }
+  const fullDecimalStakeAmount = getDecimalAmount(new BigNumber(stakeAmount), stakingToken.decimals)
+  let userNotEnoughToken
+  if (isRemovingStake) {
+    userNotEnoughToken = userData.stakedBalance.lt(fullDecimalStakeAmount)
+  } else {
+    userNotEnoughToken = userData.stakingTokenBalance.lt(fullDecimalStakeAmount)
+  }
```
Comment (hachiojidev) [inline src/views/Pools/components/PoolCard/Modals/StakeModal.tsx:85]: Ternary

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Pools/components/PoolCard/Modals/StakeModal.tsx (hunk 2 above)
Comment (memoyil) [reply src/views/Pools/components/PoolCard/Modals/StakeModal.tsx]: Done

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**